pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1619,186 +1619,9 @@ func parseConfigItems(config *zconfig.EdgeDevConfig, ctx *getconfigContext) {
-			if strings.HasPrefix(item.Value, "ssh") {
-				newGlobalConfig.SshAuthorizedKeys = item.Value
-				newGlobalConfig.SshAccess = true
-			} else {
-				err = fmt.Errorf("Invalid value for debug.enable.ssh. "+
-					"Not starting with prefix ssh. Value: %s", item.Value)
-			}
+		err = globalItemMap.parseItem(*globalConfig, item)
 
-		case "app.allow.vnc":
-			newBool, err := strconv.ParseBool(item.Value)
-			if err == nil {
-				newGlobalConfig.AllowAppVnc = newBool
-			}
-
-		case "timer.use.config.checkpoint":
-			i64, err := strconv.ParseUint(item.Value, 10, 32)
-			if err == nil {
-				newGlobalConfig.StaleConfigTime = uint32(i64)
-			}
-
-		case "timer.gc.download":
-			i64, err := strconv.ParseUint(item.Value, 10, 32)
-			if err == nil {
-				newGlobalConfig.DownloadGCTime = uint32(i64)
-			}
-
-		case "timer.gc.vdisk":
-			i64, err := strconv.ParseUint(item.Value, 10, 32)
-			if err == nil {
-				newGlobalConfig.VdiskGCTime = uint32(i64)
-			}
-
-		case "timer.gc.rkt.graceperiod":
-			i64, err := strconv.ParseUint(item.Value, 10, 32)
-			if err == nil {
-				newGlobalConfig.RktGCGracePeriod = uint32(i64)
-			}
-
-		case "timer.download.retry":
-			i64, err := strconv.ParseUint(item.Value, 10, 32)
-			if err == nil {
-				newGlobalConfig.DownloadRetryTime = uint32(i64)
-			}
-
-		case "timer.boot.retry":
-			i64, err := strconv.ParseUint(item.Value, 10, 32)
-			if err == nil {
-				newGlobalConfig.DomainBootRetryTime = uint32(i64)
-			}
-
-		case "network.allow.wwan.app.download":
-			newTs, err := types.ParseTriState(item.Value)
-			if err == nil {
-				newGlobalConfig.AllowNonFreeAppImages = newTs
-			}
-
-		case "network.allow.wwan.baseos.download":
-			newTs, err := types.ParseTriState(item.Value)
-			if err == nil {
-				newGlobalConfig.AllowNonFreeBaseImages = newTs
-			}
-
-		case "debug.default.loglevel":
-			newGlobalConfig.DefaultLogLevel = item.Value
-
-		case "debug.default.remote.loglevel":
-			newGlobalConfig.DefaultRemoteLogLevel = item.Value
-
-		case "storage.dom0.disk.minusage.percent":
-			i64, err := strconv.ParseUint(item.Value, 10, 32)
-			if err == nil {
-				newGlobalConfig.Dom0MinDiskUsagePercent = uint32(i64)
-			}
-			// Max diskspace for dom0 is 80%.
-			if newGlobalConfig.Dom0MinDiskUsagePercent > 80 {
-				err = fmt.Errorf("dom0MinDiskUsagePercent (%d) "+
-					"> maxAllowed (80). Setting it to 80.",
-					newGlobalConfig.Dom0MinDiskUsagePercent)
-				log.Errorf("parseConfigItems: %s", err)
-				newGlobalConfig.Dom0MinDiskUsagePercent = 80
-			}
-			log.Infof("Set storage.dom0MinDiskUsagePercent to %d",
-				newGlobalConfig.Dom0MinDiskUsagePercent)
-
-		case "storage.apps.ignore.disk.check":
-			newBool, err := strconv.ParseBool(item.Value)
-			if err != nil {
-				log.Errorf("parseConfigItems: bad bool value %s for %s: %s\n",
-					item.Value, key, err)
-				continue
-			}
-			newGlobalConfig.IgnoreDiskCheckForApps = newBool
-
-		default:
+		if err != nil {
 			// Handle agentname items for loglevels
```

Comment:
Thia also belongs in the parseItem. ParseItem parses both global configs as well as per-agent configs.. Move it inside parseItem